### PR TITLE
Move quote internal representations to scala.internal

### PR DESCRIPTION
**Pull request by nicolasstucki**
This object contains the internal representations of quotes and splices used in the compiler.

### CODE REVIEW

File: library/src/scala/internal/Quoted.scala [hunk 1]
```diff
@@ -1,10 +1,10 @@
-package scala.internal.quoted
+package scala.internal
```
Comment (smarter) [inline library/src/scala/internal/Quoted.scala:1]: I thought the idea was that we would no longer have scala.internal in the library and instead move it in the compiler?

Comment (nicolasstucki) [reply library/src/scala/internal/Quoted.scala:1]: This one seems to be needed in the library. User code will contain calls to these methods and those will be pickled in TASTy. It is also something that nobody should ever call directly.

Comment (smarter) [reply library/src/scala/internal/Quoted.scala:1]: Then I still think we should reconsider having it in scala.internal: this means that by default everyone has `internal` imported in their code, all for one class they cannot use.

Comment (smarter) [reply library/src/scala/internal/Quoted.scala:1]: can't this be QuotedInternal in scala.quoted or something like that?

Comment (nicolasstucki) [reply library/src/scala/internal/Quoted.scala:1]: It would make it too easy for users to access with autocomplete.

Comment (smarter) [reply library/src/scala/internal/Quoted.scala:1]: Then scala.quoted.internal or something like that so it's not polluting the top-level scala namespace. Or maybe those methods should be defined in the magic `<special-ops>` package like throw: https://github.com/lampepfl/dotty/blob/4411811afdec5db72564321d157b4775e1beb7a6/compiler/src/dotty/tools/dotc/core/Definitions.scala#L451

Comment (nicolasstucki) [reply library/src/scala/internal/Quoted.scala:1]: special ops sound like the right approach. I will try that.

**Pull request closed without merging**